src/server/dfly_main.cc

```diff
@@ -161,35 +161,39 @@ struct VersionMonitor {
   }
 
  private:
-  void RunTask(SSL_CTX* ssl_ctx);
+  struct ssl_deleter {
+    void operator()(SSL_CTX* ssl) {
+      if (ssl) {
+        free(ssl);
+      }
+    }
+  };
+  using ssl_ptr = std::unique_ptr<SSL_CTX, ssl_deleter>;
+  void RunTask(ssl_ptr);
 };
 
 void VersionMonitor::Run(ProactorPool* proactor_pool) {
   // Avoid running dev environments.
-  bool is_dev_env = false;
-  const char* env_var = getenv("DFLY_DEV_ENV");
-  if (env_var) {
+  if (getenv("DFLY_DEV_ENV")) {
     LOG(WARNING) << "Running in dev environment (DFLY_DEV_ENV is set) - version monitoring is "
                     "disabled";
-    is_dev_env = true;
   }
-  if (!GetFlag(FLAGS_version_check) || is_dev_env ||
-      // not a production release tag.
-      kGitTag[0] != 'v' || strchr(kGitTag, '-') != NULL) {
+  // not a production release tag.
+  if (!GetFlag(FLAGS_version_check) || kGitTag[0] != 'v' || strchr(kGitTag, '-')) {
     return;
   }
 
-  SSL_CTX* ssl_ctx = TlsClient::CreateSslContext();
+  ssl_ptr ssl_ctx(TlsClient::CreateSslContext());
```

Comment:
doesnt it make more sense to create the ssl cntx here and not inside RunTask?

Reply:
You mean the opposite I guess, because this is the `Run` function. I don't think it makes much sense in creating within `RunTask` because the creation can fail and launching this on a fiver (even if the overhead is small) is kinda obsolete.